client/collection.go

```diff
@@ -46,38 +46,39 @@ type Collection interface {
 	// Create a new document.
 	//
 	// Will verify the DocID/CID to ensure that the new document is correctly formatted.
-	Create(context.Context, *Document) error
+	Create(ctx context.Context, identity immutable.Option[string], doc *Document) error
```

Comment:
I do have a preference for merging this as-is, and then changing this later.  Designing an alternative flow will take thought, and be controversial, and it is not a good idea to bog down the merging of the initial PR with such a thing IMO.

We are near the start of the sprint, and this change can easily be made by any team member, in quite a short space of time (once the design is accepted by the rest of dev-db).

Reply:
@shahzadlone @jsimnz If either of you are tempted to do try and do this in this PR I strongly suggest discussing the choice to tackle it within this PR with the rest of the team first, before attempting to design it.

Reply:
My preference is based on two main criteria. The PR introduces a significant `client.Collection` breaking change (and some other interfaces) to accommodate the `identity` argument, then whatever change we implement (based on the two suggestions in #2439) will result in change many (all?) of these interfaces again. 

Additionally, and as a result there are ~400 (see pic, couple of those are comments) instances in this PR that is *just* propagating the `identity` variable around the place 
![image](https://github.com/sourcenetwork/defradb/assets/4015414/f0feb72d-7b23-4817-a1ca-ed1731d07057). 

With the `context` approach I suggested, this will basically go to zero.

Reply:
Will open a discord suggestion for this.

Reply:
Follow-up from discord discussion [here](https://discord.com/channels/427944769851752448/1200142876307374100/1222595052891799626), this will be resolved in another PR. Not a merge blocker